src/hooks/useOpenLoanHandler.tsx

```diff
@@ -106,212 +116,211 @@ export const useOpenLoanHandler = () => {
-            return
-        }
-        // TODO: wait for contract to handle collateral
-        let tx = await posManager.openPosition(
-          token0.address,
-          token1.address,
-          amt0BN,
-          amt1BN,
-          loanAmtBN,
-          accountAddress
-        )
-        let loading = notifyLoading('Waiting for block confirmation')
-        let receipt = await tx.wait()
-        if (receipt.status == 1) {
-          notifyDismiss(loading)
-          notifySuccess('Position opened successfully.')
-          return
-        }
-        notifyError('Open position was unsuccessful.')
-      } catch (e) {
-        if (typeof e === 'string') {
-          notifyError(e)
-        } else if (e instanceof Error) {
-          notifyError(e.message)
+    if (!peripheryPosManager) {
+      await getPosMgr()
+    }
+    if (peripheryPosManager) {
+      let loan = await createLoan()
+      if (loan) {
+        const { args } = loan.events[1]
+        let tokenId = args.tokenId
+        console.log(tokenId)
+        if (tokenId) {
+          let collateral = await increaseCollateral(tokenId)
+          console.log(collateral)
+          if (collateral) {
+            let borrow = await borrowLiquidity(tokenId)
+            console.log(borrow)
+            if (borrow) {
+              notifySuccess('Successful')
+            }
+          }
         }
       }
-    } else {
-      if (peripheryPosManager) {
-        let loading = notifyLoading('Waiting for block confirmation')
-        createLoan()
-          .then((res) => {
-            const { args } = res.events[1]
-            let tokenId = args.tokenId.toNumber()
-            console.log(tokenId)
-            borrowLiquidity(tokenId)
-              .then((result) => {
-                console.log(result)
-                toast.success('Open loan was successful')
-                toast.dismiss(loading)
-              })
-              .catch((err) => {
-                notifyDismiss(loading)
-                notifyError('Borrow liquidity was unsuccessful: ' + err)
-              })
-          })
-          .catch((err) => {
-            notifyDismiss(loading)
-            notifyError('Create loan was unsuccessful' + err)
-          })
-      }
     }
   }
 
   async function createLoan() {
-    if (peripheryPosManager && accountInfo) {
-      let tx = await peripheryPosManager.createLoan(
-        process.env.NEXT_PUBLIC_CFMM_ADDRESS,
-        1,
-        accountInfo?.address,
-        ethers.constants.MaxUint256
-      )
-      return await tx.wait()
-    }
-  }
-
-  async function borrowLiquidity(tokenId: number) {
-    if (peripheryPosManager && accountInfo) {
-      const BorrowLiquidityParams = {
-        cfmm: process.env.NEXT_PUBLIC_CFMM_ADDRESS,
-        protocol: 1,
-        tokenId: tokenId,
-        lpTokens: 1,
-        to: accountInfo?.address,
-        deadline: ethers.constants.MaxUint256,
+    let loading = notifyLoading('Waiting for create loan')
```

Comment:
do we need loading for the other contract calls on this page?

Reply:
Yes we are using it on three places.

1. Open loan
2. Increase collateral
3. Borrow liquidity